### PR TITLE
Add API to disable watchdog

### PR DESCRIPTION
**Pull request by tmplt**
This PR implements `embedded_hal::watchdog::WatchdogDisable` from the unproven feature set. The watchdog is enabled on reset and resets the target if not fed after approximately 15 seconds.

### CODE REVIEW

**Comment by michalfita** [comment]
This PR is based not on `develop`, but on #2, right?

**Comment by tmplt** [comment]
Correct. I'll mark it ready for review when #2 merges.

**Comment by tmplt** [comment]
I'll rebase this PR on `development` tomorrow. I thought github would be able to see past the merge commit when marking the PR ready for review, but that was not the case.

**Comment by tmplt** [comment]
There, ready for review.

File: hal/Cargo.toml [hunk 1]
```diff
@@ -16,13 +16,13 @@ keywords = [
     "hal"
 ]
 license = "0BSD"
-repository = "https://github.com/michalfita/atsams-rust"
+repository = "https://github.com/atsams-rs/atsamx7x-rust"
 readme = "README.md"
-edition = "2018"
+edition = "2021"
```
Comment (michalfita) [inline hal/Cargo.toml:21]: Are you 100% sure about that, Edition 2021 reached stability in 1.56 and somewhere earlier the Rust version was set to 1.46.

I don't mind setting minimum supported compiler @ 1.56, but I need to know there's commitment to do so.

File: hal/Cargo.toml [hunk 2]
```diff
@@ -16,13 +16,13 @@ keywords = [
     "hal"
 ]
 license = "0BSD"
-repository = "https://github.com/michalfita/atsams-rust"
+repository = "https://github.com/atsams-rs/atsamx7x-rust"
```
Comment (michalfita) [inline hal/Cargo.toml:19]: Good spot.

File: hal/Cargo.toml (hunk 1 above)
Comment (tmplt) [reply hal/Cargo.toml:21]: `rust_version = "1.46.0"`, set in the PACs define the minimum supported version. Edition is also crate-local. I do not see any issues using the latest edition in the HAL.

**Review by michalfita — APPROVED** [review]

**Pull request merged**